### DISCUSSION TITLE
iCub face expressions doesn't run

### DISCUSSION THREAD

**Discussion by pfdominey** [robotology/community#305] [Q&A] [answered]
Trying to run facial expression leds on the icub face
using yarpmanager to run
yarpdev pc104 --name /icub:face:raw --device serial --subdevice serialport --context faceExpressions --from serialport.ini

get the error

[ERR] (Face_Expressions) cannont run yarpdev on pc104: cannot run yarpdev on /pc104

note that other yarpdev can run on pc104, for the cameras, etc.

We changed the pc104 6 months ago or so.  Is there possibly a missing serial prot connection that we did not update?

thanks - cheers - Peter, faithful iCub_Lyon_01 user since 2009!

**Comment by Nicogene** [comment]
Hi @pfdominey !

It could be realted to robotology/yarp#1705, which version of yarp do you have?
Can you try to launch it from terminal and paste here the output ? Thanks!

**Comment by pfdominey** [comment, marked as answer]
It turns out that an environment variable was not set to include serial.ini
we changed that and now it works -
one more simple question
what are the command line commands to change the facial expressions?

**Comment by pattacini** [comment]
Hi @pfdominey 

Look at point 4 of this guide: http://wiki.icub.org/wiki/VVV08/face_expressions.
Typical values of `<cmd>` are: `hap`, `cun`, `evi`...

You'll find out more in [emotions.ini](https://github.com/robotology/icub-main/blob/master/app/faceExpressions/conf/emotions.ini) file.
